pkg/kubelet/cm/dra/manager.go

```diff
@@ -28,50 +28,40 @@ import (
 	"k8s.io/dynamic-resource-allocation/resourceclaim"
 	"k8s.io/klog/v2"
 	dra "k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
+// draManagerStateFileName is the file name where dra manager stores its state
+const draManagerStateFileName = "dra_manager_state"
+
 // ManagerImpl is the structure in charge of managing DRA resource Plugins.
 type ManagerImpl struct {
-	// cache contains cached claim info
-	cache *claimInfoCache
+	// checkpoint cache contains cached claim info
+	cache state.CheckpointState
 
 	// KubeClient reference
 	kubeClient clientset.Interface
 }
 
 // NewManagerImpl creates a new manager.
-func NewManagerImpl(kubeClient clientset.Interface) (*ManagerImpl, error) {
+func NewManagerImpl(kubeClient clientset.Interface, stateFileDirectory string) (*ManagerImpl, error) {
 	klog.V(2).InfoS("Creating DRA manager")
 
-	manager := &ManagerImpl{
-		cache:      newClaimInfoCache(),
-		kubeClient: kubeClient,
-	}
-
-	return manager, nil
-}
-
-// Generate container annotations using CDI UpdateAnnotations API.
-func generateCDIAnnotations(
-	claimUID types.UID,
-	driverName string,
-	cdiDevices []string,
-) ([]kubecontainer.Annotation, error) {
-	annotations, err := updateAnnotations(map[string]string{}, driverName, string(claimUID), cdiDevices)
+	stateImpl, err := state.NewCheckpointState(stateFileDirectory, draManagerStateFileName)
 	if err != nil {
-		return nil, fmt.Errorf("can't generate CDI annotations: %+v", err)
+		klog.ErrorS(err, "Could not initialize checkpoint manager, please drain node and remove dra state file")
+		return nil, err
 	}
 
```

Comment:
If we did as suggested above, then it would be here that I would imagine we take the state we just pulled, and we update the cache with both it, and any derived information before building the manager below.